### ISSUE TITLE
Add setup to make more realistic tests

### ISSUE THREAD

**Issue by MatheusRich** [MichaelSp/rspec_vs_minitest_vs_cucumber#4]
The current tests compare the raw performance of the test libraries.

This is nice, but it's quite distance from a realistic test suite.

Could we add some setup to simulate an IO/DB call?

Maybe just a `sleep 0.001` would do?

**Comment by MichaelSp** [comment]
That's right and also what I wanted to achieve. The tests are not supposed to be realistic, but just to compare the performance of the framework itself.
From my point of view it doesn't make any difference what payload to test if comparing test frameworks as long as the payload is all same.
Adding sleep or any other implementation just adds complexity to the test setup and has to be the same for all frameworks. So not sure what added value that has.

If I choose a framework I want to know two things:
 1. How do I use it? 
 2. Is it fast?


Happy to discuss, though...

**Comment by MatheusRich** [comment]
I don't know if I was clear, but my point is to show that even though minitest is much faster than Rspec, that has a lower impact in a real test suite. 

You can see [here](https://speakerdeck.com/palkan/rubyconfby-minsk-2017-run-test-run?slide=31) that Vladimir measured only 10% difference in speed between minitest and rspec.

I see that 10% is still nice speedup, but I think an _additional_ test suite with some IO could help show something more interesting.

**Comment by MichaelSp** [comment]
I agree, a total run-time diff of 10% is negligible. 
But this repo is not about showing interesting use cases for test-frameworks. It's just about comparing the raw performance. There might be better ways to do so and I'm always open to improve. 
BTW: I don't claim that this is a real-world-scenario.

There might be more interesting and more real-world-like benchmarks out there. But from my point of view they are adding more noise to the numbers.

However: If you want to contribute more realistic scenarios, feel free to send me PR's (maybe in subfolders). But I'd like to keep on very basic and simple scenario like the current one.
